lib/widgets/payment_request_dialog.dart

```diff
@@ -20,48 +27,237 @@ class PaymentRequestDialog extends StatefulWidget {
     return PaymentRequestDialogState();
   }
 }
 
-class PaymentRequestDialogState extends State<PaymentRequestDialog> {
+class PaymentRequestDialogState extends State<PaymentRequestDialog>
+    with SingleTickerProviderStateMixin {
+
   final _formKey = GlobalKey<FormState>();
   TextEditingController _invoiceAmountController = new TextEditingController();
 
+  AnimationController controller;
+  Animation<double> opacityAnimation;
+  Animation<double> borderAnimation;
+  Animation<RelativeRect> transitionAnimation;
+  Animation<Color> colorAnimation;
+
+  PaymentRequestState _state;
+
+  Int64 _amountToPay;
+  String _amountToPayStr;
+
+  bool _inProgress = false;
+  bool _isInit = false;
+
   @override
-  void initState() {    
+  void initState() {
     super.initState();
-    _invoiceAmountController.addListener((){
+    widget.accountBloc.accountStream.listen((acc) {
+      if (acc.paymentRequestInProgress != null && acc.paymentRequestInProgress.isNotEmpty) {
+        _inProgress = true;
+      } else if (acc.paymentRequestInProgress == null || acc.paymentRequestInProgress.isEmpty){
+        _inProgress = false;
+      }
+    });
+    _listenPaymentsResults();
+    setState(() {
+      _state = PaymentRequestState.PAYMENT_REQUEST;
+    });
+    controller = AnimationController(
+        vsync: this, duration: Duration(milliseconds: 600));
+    borderAnimation = Tween<double>(begin: 0.0, end: 8.0).animate(
+        CurvedAnimation(parent: controller, curve: Curves.fastOutSlowIn));
+    opacityAnimation = Tween<double>(begin: 0.0, end: 1.0).animate(
+        CurvedAnimation(parent: controller, curve: Curves.fastOutSlowIn));
+    colorAnimation = new ColorTween(
+      begin: theme.BreezColors.blue[500],
+      end: theme.BreezColors.white[500],
+    ).animate(controller)
+      ..addListener(() {
+        setState(() {});
+      });
+    controller.addStatusListener((status) {
+      if (status == AnimationStatus.dismissed) {
+        Navigator.pop(context);
+      }
+    });
+    _invoiceAmountController.addListener(() {
       setState(() {});
     });
   }
 
+  @override
+  void didChangeDependencies() {
+    if (!_isInit) {
+      var _yMargin = (MediaQuery.of(context).size.height - 300.0) / 2;
+      if (widget.invoice.payeeImageURL.isNotEmpty){
+        _yMargin = _yMargin - 10;
+        if (widget.invoice.description.isNotEmpty){
+          var lineCount = (widget.invoice.description.length / 30).ceil();
+          _yMargin = _yMargin - (lineCount*24);
+        }
+      }
+      var _paymentItemStartPosition = ((MediaQuery.of(context).size.height) * 37)/100;
+      var _paymentItemEndPosition = ((MediaQuery.of(context).size.height) * 50)/100;
+      transitionAnimation = new RelativeRectTween(
+          begin: new RelativeRect.fromLTRB(0.0, _paymentItemStartPosition, 0.0, _paymentItemEndPosition),
+          end: new RelativeRect.fromLTRB(32.0, _yMargin, 32.0, _yMargin))
+          .animate(controller);
```

Comment:
I also think we need a clever solution here. I'm not really comfortable with this.
We are controlling the dialog size with transitionAnimation here. 300.0 = base dialog height and we add 10.0 if there's payee image and 24.0 per line if that request has note.
- We should learn the size of the dialog if it was rendered with IntrinsicHeight and use that instead of 300.0 without the payeeImage/description calculations.

 Start and end position of payment item is an estimate. 
- We should scroll to top of payment list after successful payment and use the exact spot of first payment items start and end position.

Reply:
@erdemyerebasmaz  I share the same concerns with you...
we can't use relative height such as: `((MediaQuery.of(context).size.height) * 37)/100;` as it will land on different points in different resolutions.
The position in  Y axis of where the first payment item starts is the same on every device and we can pass it to the widget if needed.
I am suggesting the following simple approach to start with:

1.  Let's move all the animation initialization to the fulfilledPayments listener. This way we can safely query the height of the widget and know for sure it was fully rendered. 
2. for the animation we can find the dialog position and sizes using `context.findRenderObject`. For the end state of the animation I suggest to leave the Y Axis position as in the start state and only shrink the height. Let's see how this looks before we struggle to pass the exact position of the first item...
I think it will a lot simplify the code and take out all the calculations.